MediaBrowser.Providers/Manager/ProviderManager.cs

```diff
@@ -150,52 +162,84 @@ public Task<ItemUpdateType> RefreshSingleItem(BaseItem item, MetadataRefreshOpti
+                    }
                 }
-                else
+
+                // TVDb will sometimes serve a rubbish 404 html page with a 200 OK code, because reasons...
+                if (contentType.Equals(MediaTypeNames.Text.Html, StringComparison.OrdinalIgnoreCase))
                 {
-                    throw new HttpRequestException("Invalid image received: contentType not set.", null, response.StatusCode);
+                    throw new HttpRequestException("Invalid image received.", null, HttpStatusCode.NotFound);
                 }
-            }
 
-            // TVDb will sometimes serve a rubbish 404 html page with a 200 OK code, because reasons...
-            if (contentType.Equals(MediaTypeNames.Text.Html, StringComparison.OrdinalIgnoreCase))
-            {
-                throw new HttpRequestException("Invalid image received.", null, HttpStatusCode.NotFound);
-            }
+                // some iptv/epg providers don't correctly report media type, extract from url if no extension found
+                if (string.IsNullOrWhiteSpace(MimeTypes.ToExtension(contentType)))
+                {
+                    // Strip query parameters from url to get actual path.
+                    contentType = MimeTypes.GetMimeType(new Uri(url).GetLeftPart(UriPartial.Path));
+                }
 
-            // some iptv/epg providers don't correctly report media type, extract from url if no extension found
-            if (string.IsNullOrWhiteSpace(MimeTypes.ToExtension(contentType)))
-            {
-                contentType = MimeTypes.GetMimeType(url);
-            }
+                if (!contentType.StartsWith("image/", StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new HttpRequestException($"Request returned {contentType} instead of an image type", null, HttpStatusCode.NotFound);
+                }
 
-            var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
-            await using (stream.ConfigureAwait(false))
-            {
-                await SaveImage(
-                    item,
-                    stream,
-                    contentType,
-                    type,
-                    imageIndex,
-                    cancellationToken).ConfigureAwait(false);
+                var stream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
+                await using (stream.ConfigureAwait(false))
+                {
+                    var memoryStream = new MemoryStream();
+                    await using (memoryStream.ConfigureAwait(false))
+                    {
+                        await stream.CopyToAsync(memoryStream, cancellationToken).ConfigureAwait(false);
+                        memoryStream.Seek(0, SeekOrigin.Begin);
+                        _memoryCache.Set(url, (contentType, memoryStream.GetBuffer()), TimeSpan.FromSeconds(10));
```

Comment:
`MemoryStream.GetBuffer()` returns the underlying buffer, it's length may not match the actual content
I'd use `HttpContent.ReadAsByteArrayAsync` and construct a `MemoryStream` from that for use by `SaveImage`